doc/api/diagnostics_channel.md

```diff
@@ -409,6 +459,538 @@ channel.subscribe(onMessage);
 channel.unsubscribe(onMessage);
 ```
 
+#### `channel.bindStore(store[, transform])`
+
+<!-- YAML
+added:
+ - REPLACEME
+-->
+
+> Stability: 1 - Experimental
+
+* `store` {AsyncLocalStorage} The store to which to bind the event data
+* `transform` {Function} Transform event data before setting the store context
+
+When [`channel.runStores(data, ...)`][] is called, the given event data will
+be applied to any store bound to the channel. If the store has already been
+bound the previous `transform` function will be replaced with the new one.
+The `transform` function may be omitted to have the event data be used as the
+context directly.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+import { AsyncLocalStorage } from 'node:async_hooks';
+
+const store = new AsyncLocalStorage();
+
+const channel = diagnostics_channel.channel('my-channel');
+
+channel.bindStore(store, (data) => {
+  return { data };
```

Comment:
Isn't a transform function like this - which returns a new object - problematic in the e.g. the `traceCallback` api? One might expect to get the same `context` object in all events within`traceCallback`.
If I follow the code correct a fresh object will be created each time.

Or is this intended and user has to take care to do the right thing.

Reply:
It's only used for the store replacement, not for the event, and the store replacement will only happen if you've explicitly done a bindStore to that specific event with your store. So if you _want_ to have a new context object in the callback you could, but it won't happen unless you tell it to do that. The intent is more that you would store your span object on the trace context object in the transform for the start and then optionally you could restore that as the context by retrieving it in the transform for the callback.

My feeling on it is that it's maybe a moderately helpful escape hatch in the event the context does not propagate the way you think it should, but it's basically just a way to recover a broken propagation that should probably be fixed _elsewhere_.

Reply:
Also, an example of the context restoration use case, for clarity:

```js
startChannel.bindStore(spanStore, (ctx) => {
  const parentSpan = spanStore.getStore()
  const span = new Span(ctx, parentSpan)
  ctx.span = span
  return span
})

asyncStartChannel.bindStore(spanStore, (ctx) => {
  // Context could have broken before asyncStart was reached,
  // but we stored the span on ctx so we can restore from that.
  return ctx.span
})
```

Reply:
ok thanks. got it (once again :o)).
seems there are too many contexts around for me to stay on track...